src/test/java/com/g4s8/ghman/web/PsUserByIdTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2019 Kirill Ch. (g4s8.publci@gmail.com)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to read
+ * the Software only. Permissions is hereby NOT GRANTED to use, copy, modify,
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.g4s8.ghman.web;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import javax.ws.rs.HttpMethod;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.utils.URIBuilder;
+import org.cactoos.text.FormattedText;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.takes.Response;
+import org.takes.facets.auth.Identity;
+import org.takes.rq.RqFake;
+import org.takes.rs.RsWithStatus;
+
+/**
+ * Test for {@link PsUserById}.
+ * @since 1.0
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+class PsUserByIdTest {
+
+    @Test
+    void returnsIdentityOnEnter() throws IOException, URISyntaxException {
```

Comment:
@olenagerasimova also add a test for the case when the answer is not present

Reply:
@victornoel you mead when required request parameter is absent?

Reply:
@olenagerasimova yes, the method enter returns an `Opt` return type, so it means there are situations when this must return an empty `Opt`. This should be tested too so that we don't find ourselves with a broken implementation that returns a value when it shouldn't.

Reply:
@victornoel when does the method we are testing here returns empty `Opt`? It either returns not empty `Opt` or (if required request parameter is absent) throws an exception

Reply:
@olenagerasimova indeed, I don't know if this is expected that it doesn't return an empty Opt instead... there is something strange in the behaviour of this object, luckily the tests should help us clarify the situation. @g4s8 what should happen here?

Reply:
@g4s8 ping

Reply:
@olenagerasimova @victornoel `Pass` should return empty `Opt` if it can't process this kind of request, e.g. if we have few auth methods, one by cookie value and one by API key, and we have two `Pass` implementation for each method, then `PsByApiKey` should return empty `Opt` for request without API key in headers but with auth cookie